/*
 * Copyright (c) 2017 Broadcom. The term "Broadcom" refers
 * to Broadcom Limited and/or its subsidiaries.
 
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 
 * You may obtain a copy of the License at
 * http://www.apache.org/licenses/LICENSE-2.0
 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
 
syntax = "proto2";

import "sai_tam_main.proto";
import "sai_tam_buffer_stats.proto";

enum ThresholdBreachSourceType {
    THRESHOLD_BREACH_AT_QUEUE = 1;
    THRESHOLD_BREACH_AT_GLOBAL_POOLS = 2;
    THRESHOLD_BREACH_AT_IPG = 3;
    THRESHOLD_BREACH_AT_PORT_POOL = 4;
}

message ThresholdSource {
    required ThresholdBreachSourceType type = 1;
    optional uint64 port_oid = 2;
    optional QueueType queue_type = 3;    
    optional uint64 queue_oid = 4;
    optional BufferPoolType pool_type = 5;    
    optional uint64 pool_oid = 6;
    optional IPGType ipg_type = 7;
    optional uint64 ipg_oid = 8;
}
    
message ThresholdBreach {
    required ThresholdSource breach_source = 1;
    optional SwitchBufferStats buffer_stats = 2;
}

